content/docs/10.plugin-developer-guide/07.outputs.md

```diff
@@ -2,40 +2,98 @@
+- `{}` can be used for single-line JSON objects.
 - `::{}::` can be used for multi-line JSON objects.
 
-Here is the representation of the output object, it's a map in which you can add any key/value pair:
+Below is an example of an output object. It's a **dictionary** object that you can use to add any key-value pairs:
 
-```json5
-"outputs": { // map of key value with all the outputs you want to send
+```json
+"outputs": { // map/dicttionary of key-value pairs with outputs
     "my-key": "my-value",
-    "my-list": [1, 2, 3] // you can use  json type (bool, array, map, …)
+    "my-list": [1, 2, 3] // you can use  a variety of data types supported in a JSON request payload
 }
 ```
 
-Here is the representation of the metrics object, it's a list of JSON objects:
-```json5
-"metrics": [ // you can send multiple metrics at once
+Here is the representation of a `metrics` object. It's a **list of dictionaries**:
+```json
+"metrics": [ // you can send multiple metrics at once - each is a dictionary within a list
     {
         "name": "my-counter", // mandatory, the name of the metrics
-        "type": "counter", // mandatory, "counter" or "timer"
-        "value": 1.2, // mandatory (double), counter to add, or duration in seconds for timer
-        "tags":{ // optional list of tags that will expose internal details
-          "type":"read",
-          "location":"eu"
+        "type": "counter", // mandatory, "counter" or "timer" metric type
+        "value": 1.2, // mandatory (double), counter to measure number of objects/rows/items processed in a given task, or duration in seconds for the timer metric
+        "tags": { // optional list of tags that will expose internal details
+          "type": "read",
+          "location": "EU"
         }
     }
 ]
 ```
 
+## When to use metrics and when to use outputs?
+
+### Use cases for outputs
+
+Outputs are task-run artifacts. They are generated as a result of a given task. If you want to track task-run metadata across multiple executions of a flow, and this metadata is of an arbitrary data type (it might be a string, integer, ...), use `outputs` rather than `metrics`. 
+
+Examples of metadata you may want to track as `outputs`: 
+
+- the **number of rows** processed in a given task that you want to use in subsequent tasks validating the number of rows, 
+- the **accuracy score** of a trained ML model in order to compare this result (output artifact) across multiple workflow executions, 
+- other pieces of **metadata** you want to track across executions of a flow (e.g. a dataset name used within a Python ETL script).
+
+### Use cases for metrics
+
+Metrics are intended to track custom numeric (metric type: `counter`) or duration (metric type: `timer`) attributes that you may want to visualize across tasks. Metrics are typically expressed as `double` data type numerical values.
```

Comment:
```suggestion
Metrics are intended to track custom numeric (metric type: `counter`) or duration (metric type: `timer`) attributes that you may want to visualize across flow executions. Metrics are typically expressed as `double` data type numerical values.
```

Reply:
it's both right?

across task runs and flow executions

Reply:
Yes, but I see metrics more for analytics of tasks metadata so looking at different values of the same metric on different executions instead of values of different metrics in the same flow execution.

The metrics dashboard will be at the flow level and will enable selecting a metrics of a task and graph the metric with the values of all flow executions.

Reply:
Correct, it's on a flow execution level, but in this partition example, we also compare times between mapped child tasks (partitions) so comparing metrics across tasks seems possible

if you feel strongly about excluding tasks here, LMK

Reply:
You're the leader on everything documentation related so I trust you on this ;)

Both is OK for me as you said.